src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3758,6 +3809,67 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _showContextMenuAt(_toControlOrigin(cursorPos));
     }
 
+    double TermControl::CalculateQuickFixButtonWidth()
+    {
+        return GetPadding().Left;
+    }
+
+    double TermControl::CalculateQuickFixButtonCollapsedWidth()
+    {
+        return GetPadding().Left / 3.0;
+    }
+
+    void TermControl::ShowQuickFixMenu()
+    {
+        if (!_quickFixesAvailable)
+        {
+            QuickFixButton().Visibility(Visibility::Collapsed);
+            return;
+        }
+
+        auto quickFixBtn{ QuickFixButton() };
+
+        // If the gutter is narrow, display the collapsed version
+        const auto& termPadding = GetPadding();
+
+        _quickFixButtonCollapsible = termPadding.Left < CharacterDimensions().Width;
+        VisualStateManager::GoToState(*this, !_quickFixButtonCollapsible ? StateNormal : StateCollapsed, false);
+
+        const auto rd = get_self<ControlCore>(_core)->GetRenderData();
+        rd->LockConsole();
+        const auto viewportBufferPosition = rd->GetViewport();
+        const auto cursorBufferPosition = rd->GetCursorPosition();
+        rd->UnlockConsole();
+        if (cursorBufferPosition.y < viewportBufferPosition.Top() || cursorBufferPosition.y > viewportBufferPosition.BottomExclusive())
+        {
+            quickFixBtn.Visibility(Visibility::Collapsed);
+            return;
+        }
+
+        // draw the button in the gutter
+        const auto& cursorPosInDips = CursorPositionInDips();
+        Controls::Canvas::SetLeft(quickFixBtn, -termPadding.Left);
+        Controls::Canvas::SetTop(quickFixBtn, cursorPosInDips.Y);
+        quickFixBtn.Visibility(Visibility::Visible);
+    }
+
+    void TermControl::_bubbleSearchMissingCommand(const IInspectable& /*sender*/, const Control::SearchMissingCommandEventArgs& args)
+    {
+        _quickFixesAvailable = true;

```

Comment:
Should we hold off on `_quickFixesAvailable = true` until after we get the `UpdateWinGetSuggestions` callback?

Reply:
Good call. Actually even removed `_quickFixesAvailable` and made it an API off of `ControlCore`. More future-proofed that way.